src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMissingWhitespaceAfterAsteriskCheck.java

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * <p>
+ * Checks that there is at least one whitespace after the leading asterisk.
+ * Although spaces after asterisks are optional in the Javadoc comments, their absence
+ * makes the documentation difficult to read. It is the de facto standard to put at least
+ * one whitespace after the leading asterisk.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
+ * Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocMissingWhitespaceAfterAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Code Example:
+ * </p>
+ * <pre>
+ * &#47;** This is valid single-line Javadoc. *&#47;
+ * class TestClass {
+ *   &#47;**
+ *     *This is invalid Javadoc.
+ *     *&#47;
+ *   int invalidJavaDoc;
+ *   &#47;**
+ *     * This is valid Javadoc.
+ *     *&#47;
+ *   void validJavaDocMethod() {
+ *   }
+ *   &#47;**This is invalid single-line Javadoc. *&#47;
+ *   void invalidSingleLineJavaDocMethod() {
+ *   }
+ *   &#47;** This is valid single-line Javadoc. *&#47;
+ *   void validSingleLineJavaDocMethod() {
+ *   }
+ * }
+ * </pre>
+ *
+ * @since 8.31
+ */
+@StatelessCheck
+public class JavadocMissingWhitespaceAfterAsteriskCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_KEY = "javadoc.missing.whitespace";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+            JavadocTokenTypes.LEADING_ASTERISK,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode detailNode) {
+        final DetailNode textNode;
+
+        if (detailNode.getType() == JavadocTokenTypes.JAVADOC) {
+            textNode = JavadocUtil.getFirstChild(detailNode);
+        }
+        else {
+            textNode = JavadocUtil.getNextSibling(detailNode);
+        }
+
+        if (textNode != null
+                && textNode.getType() != JavadocTokenTypes.EOF
+                && !hasWhitespaceAfterAsterisk(textNode)) {
+            log(textNode.getLineNumber(), textNode.getColumnNumber(), MSG_KEY);
+        }
+    }
+
+    /**
+     * Checks if there is at least one whitespace after leading asterisks.
+     *
+     * @param node the node after the leading asterisks.
+     * @return true if there is at least one white space after asterisk.
+     */
+    private static boolean hasWhitespaceAfterAsterisk(DetailNode node) {
+        boolean hasWhitespaceAfterAsterisk = true;
```

Comment:
please change a logic a bit.
At the beginning we do not know that we have whitespace so `hasWhitespaceAfterAsterisk = false;` if you find it you do `hasWhitespaceAfterAsterisk = true;`

Reply:
I set the initial value to be true because we wish to allow comment like
```
 /***   textNode = '*'
  * Something
  */
```
So if the node text is a pure sequence of '\*' I want to return true. If the initial value is set to be false, I will need additional code to perform this check. Is this ok? Or I actually should write another method to check this?

Reply:
You should never "lie" in code. If you not sure - you are not sure.
At the beginning is in FALSE, if met space it become - true. If no characters other then "*" spacial state that you should consider same as there is space - it has to written in code. Nobody need to remember this.
name method `hasWhitespaceAfterAsteriskBeforeText` and write logic to follow a name.

Reply:
done